edx_django_utils/cache/utils.py

```diff
@@ -341,6 +346,21 @@ def __eq__(self, other):
         else:
             return self.key == other.key  # cache misses have no value attribute
 
+    def __hash__(self):
+        return hash(self.key)
+
     def __ne__(self, other):
         """Overrides the default implementation (unnecessary in Python 3)"""
         return not self.__eq__(other)
+
+
+def get_user(user_id):
+    """
+    Return the user object from the request cache
+    """
+    key = 'auth_user.%s' % user_id
```

Comment:
I see. It is a little confusing because the name of the default namespace includes `edx_django_utils` in the name to help avoid collisions, while at the same time, the app that is using that RequestCache really owns that namespace and shouldn't have to be concerned with possible key collisions within the default.  (Is that just more confusing?)

I think we should probably differentiate between the default and a namespace for any keys this repo wants to create.  For example: 
```
DJANGO_UTILS_REQUEST_CACHE_NAMESPACE = '{}.internal'.format(DEFAULT_NAMESPACE)
```

I know it is unlikely that another app will create a key of `'auth_user.%s' % user_id` in the default space, but if they did, it would not be a problem.  Does this make sense?